app-modules/in-app-communication/resources/js/TipTap/Extentions/SafeLink.js

```diff
@@ -0,0 +1,55 @@
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+import Link from '@tiptap/extension-link';
+import { mergeAttributes } from '@tiptap/core';
+
+export const SafeLink = Link.extend({
+    renderHTML({ HTMLAttributes }) {
+        // This is directly pulled from the Link extension - leave as is.
+        // eslint-disable-next-line no-script-url
+        if (HTMLAttributes.href?.startsWith('javascript:')) {
+            return [
+                'button',
+                mergeAttributes(this.options.HTMLAttributes, { ...HTMLAttributes, 'data-safe-link': 'true', href: '' }),
+                0,
+            ];
+        }
```

Comment:
So does this prevent links where the href contains `javascript:`?

It would result in them having empty `href` values?

If so, is there a UX issue there that a user would think that it is broken?

Reply:
This was pulled directly from the `link-extension` that we were previously using, so functionality will remain unchanged. https://github.com/ueberdosis/tiptap/blob/main/packages/extension-link/src/link.ts#L128

Reply:
@dgoetzit sounds good, if it becomes confusing for users later we can update. I doubt we will even have many instances of someone trying to send a `javascript:` link.

If you can just resolve the security issue in SonarCloud by marking it as safe and providing a comment on why it is so.